### DISCUSSION TITLE
MATEK F411TE - Can not enable GPS

### DISCUSSION THREAD

**Discussion by TrueKillRob** [iNavFlight/inav#8246] [Q&A] [answered]
INAV Version: 5.0.0
Controller: MATEK F411TE
GPS: TBS M8-2

When I enable GPS in the Configuration-Tab it is disabled again after "Save and Reboot"
When I confugre UART1, where the GPS is connected, as GPS, the UART is completly restored to default after "Save and Restore"
When I configure on the GPS-Tab (Enabling UBLOX or UBLOX7 ...) it is disabled after "Save and Restore"

Thank you for any Help
Robert

**Comment by erstec** [comment]
> INAV Version: 5.0.0 Controller: MATEK F411TE GPS: TBS M8-2
> 
> When I enable GPS in the Configuration-Tab it is disabled again after "Save and Reboot" When I confugre UART1, where the GPS is connected, as GPS, the UART is completly restored to default after "Save and Restore" When I configure on the GPS-Tab (Enabling UBLOX or UBLOX7 ...) it is disabled after "Save and Restore"
> 
> Thank you for any Help Robert

You need to disable all other things on UART1, like MSP, Serial RX and etc.
It works as should, I've just tested it on same FC.

**Comment by TrueKillRob** [comment]
I tried this but still the complete tab will be reset on "Save and Reboot".
Now I use SoftSerial 1. Here the GPS works. Do you know how high the impact for the CPU is and does it help to reduce the baud rate?

Thanks

**Reply by erstec** [reply]
I don't think it is a MCU load thning.
Did you selected Full chip erase during firmware flashing?
Which version of INAV Configurator you are using?
Please show 'diff all' of your current configuration (before you trying to apply mentioned changes in Ports tab).

**Comment by TrueKillRob** [comment]
Hello,

now I flashed the FC again and did a full chip erase.
**It works now!!!**
I can use the UART1 for the GPS and so I dont need to use the SoftUart 

Thank's to all for this grat support
Robert

BTW: I use the INAV Configrator 5.0.0
...here is the 'diff all':

`diff all

# version
# INAV/MATEKF411TE 5.0.0 Jun 20 2022 / 11:11:14 (d2adf35c)
# GCC-10.2.1 20201103 (release)

# start the command batch
batch start

# reset configuration to default settings
defaults noreboot

# resources

# mixer

mmix reset

mmix 0  1.000  0.000  0.000  0.000

# servo mix
smix reset

smix 0 1 0 -100 0 -1
smix 1 2 1 100 0 -1
smix 2 2 2 -100 0 -1
smix 3 3 1 -100 0 -1
smix 4 3 2 -100 0 -1

# servo

# safehome

# logic

# gvar

# pid

# feature
feature MOTOR_STOP
feature GPS
feature PWM_OUTPUT_ENABLE

# beeper

# blackbox
blackbox -NAV_ACC
blackbox NAV_POS
blackbox NAV_PID
blackbox MAG
blackbox ACC
blackbox ATTI
blackbox RC_DATA
blackbox RC_COMMAND
blackbox MOTORS

# map
map TAER

# serial
serial 30 2 115200 115200 0 115200
serial 31 4096 115200 115200 0 115200

# led

# color

# mode_color

# aux
aux 0 0 2 1300 1700
aux 1 1 3 1000 1300
aux 2 2 3 1300 1700
aux 3 3 1 1300 1700
aux 4 5 2 1300 1700
aux 5 36 1 1700 2000

# adjrange

# rxrange

# temp_sensor

# wp
#wp 0 invalid

# osd_layout
osd_layout 0 0 26 2 V
osd_layout 0 1 1 12 V
osd_layout 0 3 8 6 V
osd_layout 0 7 12 14 V
osd_layout 0 9 1 6 V
osd_layout 0 11 2 13 V
osd_layout 0 12 1 14 V
osd_layout 0 13 2 7 V
osd_layout 0 14 0 1 V
osd_layout 0 15 22 8 V
osd_layout 0 20 17 1 V
osd_layout 0 21 6 1 V
osd_layout 0 22 20 3 V
osd_layout 0 23 22 3 V
osd_layout 0 28 23 14 V
osd_layout 0 30 1 15 V
osd_layout 0 34 10 3 V
osd_layout 0 36 2 14 H
osd_layout 0 86 0 2 V
osd_layout 0 120 24 12 V

# master
set gyro_main_lpf_hz = 25
set dynamic_gyro_notch_q = 250
set dynamic_gyro_notch_min_hz = 30
set gyro_zero_y = -4
set gyro_zero_z = -2
set ins_gravity_cmss =  980.953
set acc_hardware = BMI270
set acczero_x = -16
set acczero_y = 26
set acczero_z = 34
set accgain_x = 4080
set accgain_y = 4093
set accgain_z = 4079
set rangefinder_hardware = VL53L0X
set align_mag = CW270FLIP
set mag_hardware = NONE
set baro_hardware = SPL06
set serialrx_provider = SRXL2
set motor_pwm_protocol = STANDARD
set platform_type = AIRPLANE
set model_preview_type = 29
set small_angle = 180
set imu_acc_ignore_rate = 9
set imu_acc_ignore_slope = 5
set applied_defaults = 3
set gps_provider = UBLOX7
set gps_sbas_mode = EGNOS
set gps_ublox_use_galileo = ON
set airmode_type = STICK_CENTER_ONCE
set nav_wp_radius = 5000
set nav_rth_allow_landing = FS_ONLY
set nav_rth_altitude = 5000
set nav_fw_control_smoothness = 2
set osd_video_system = PAL
set osd_time_alarm = 0
set tz_offset = 60
set tz_automatic_dst = EU
set vtx_band = 4
set vtx_power = 2
set vtx_low_power_disarm = ON
set dji_use_name_for_messages = OFF

# profile
profile 1

set fw_p_pitch = 15
set fw_i_pitch = 5
set fw_d_pitch = 5
set fw_ff_pitch = 80
set fw_p_roll = 15
set fw_i_roll = 3
set fw_d_roll = 7
set fw_p_yaw = 20
set fw_i_yaw = 0
set fw_ff_yaw = 100
set max_angle_inclination_rll = 450
set dterm_lpf_hz = 10
set fw_turn_assist_pitch_gain =  0.500
set nav_fw_pos_z_p = 15
set nav_fw_pos_z_d = 5
set nav_fw_pos_xy_p = 60
set d_boost_min =  1.000
set d_boost_max =  1.000
set rc_expo = 30
set rc_yaw_expo = 30
set roll_rate = 18
set pitch_rate = 9
set yaw_rate = 3

# profile
profile 2


# profile
profile 3


# battery_profile
battery_profile 1

set throttle_idle =  5.000

# battery_profile
battery_profile 2


# battery_profile
battery_profile 3


# restore original profile selection
profile 1
battery_profile 1

# save configuration
save

# `

**Comment by erstec** [comment, marked as answer]
@TrueKillRob so Full chip erase do the trick. Perfect.
